index.js

```diff
@@ -199,10 +200,36 @@ module.exports = function(version, _options) {
 
             return this.tokenize(language, instruction, replaceTokens);
         },
+        grammarize: function(language, name, grammar) {
+            // Process way/rotary name with applying grammar rules if any
+            if (name && grammar && grammars && grammars[language] && grammars[language][version]) {
+                var rules = grammars[language][version][grammar];
+                if (rules) {
+                    // Pass original name to rules' regular expressions enclosed with spaces for simplier parsing
+                    var n = ' ' + name + ' ';
```

Comment:
It's currently possible for clients to manipulate a step's name before compilation. For example, the Mapbox Directions API currently inserts SSML markup around numbers within the name so that speech synthesizers pronounce them more casually. I think that could potentially interfere with this feature. That would be a good argument for implementing #52, so that clients can reliably manipulate the name after grammaticalization.

Reply:
No problem, if clients will not damage `{way_name:*}` vars - for example, ["Restore names highlighting" PR](https://github.com/Project-OSRM/osrm-frontend/pull/259) adds HTML tags around these vars and all works as on snapshot above.
But if Mapbox Directions inserts SSML tags right instead `{way_name}`, this looks wrong - for example, Nuance uses [`\tn=address\` tags](https://github.com/yuryleb/garmin-russian-tts-voices/blob/master/src/Pycckuu__Milena%202.10/RULESET.TXT#L141) in its Vocalizer voices to specify start of address text specially to pronounce numbers and known street abbreviations as address but doesn't change original address/name text.

Reply:
But actually it's no problem, even if Mapbox will catch and replace `{way_name}` and skips `{way_name:accusative}` etc - this allows further "grammarization" to work :wink: SSML tags actually important only for English addresses, Russian & many others have no special requirements for them.

Reply:
I’m referring to a JavaScript port of mapbox/mapbox-navigation-ios#552 for the Directions API’s (not yet documented) `voice_instructions` parameter. Essentially we’re working around a problem where Amazon Polly assumes an address is abbreviated, whereas OSM never abbreviates road names. The workaround is to wrap only numbers in `<say-as interpret-as="address">` tags, since we still want numbers to be optimized for speech even if the rest of the words can’t be treated as part of an address. For the JavaScript port, we’re changing the value of `way_name` itself, so that by the time `compile()` runs, `way_name` contains SSML code. 😬

I’ll be the first to admit it’s a hack, but I wanted to point this out because clients may be inclined to do similar things to the value of `way_name` unless we advise against doing so in this project’s documentation. If this PR forces us to implement #52, that’d be a wonderful outcome in my opinion. 😉